vmm/src/lib.rs

```diff
@@ -255,16 +255,13 @@ impl Vmm {
         // Without ACPI, a reset is equivalent to a shutdown
         #[cfg(not(feature = "acpi"))]
         {
-            if let Some(ref mut vm) = self.vm {
-                vm.shutdown()?;
-                return Ok(());
-            }
+            return self.vm_shutdown()?;
```

Comment:
2 things:
* You don't want the `?`
* You need to guard that with ` if self.vm.is_some() {` to trick clippy:

```patch
diff --git a/vmm/src/lib.rs b/vmm/src/lib.rs
index 75f3653..fb54f68 100644
--- a/vmm/src/lib.rs
+++ b/vmm/src/lib.rs
@@ -255,9 +255,8 @@ impl Vmm {
         // Without ACPI, a reset is equivalent to a shutdown
         #[cfg(not(feature = "acpi"))]
         {
-            if let Some(ref mut vm) = self.vm {
-                vm.shutdown()?;
-                return Ok(());
+            if self.vm.is_some() {
+                return self.vm_shutdown();
             }
         }

```

Reply:
I do not add if condition for no acpi branch originally, since self.vm_shutdown() already has this judgement, but cargo rustc --bin cloud-hypervisor --no-default-features --features "pci" -- -D warnings will report a "unreachable statement" warning for the following branch. I add the if back to remove the warning